### PR TITLE
Install a TraceListener for integration testing

### LINKED ISSUE #33599: Debug assertions cause integration test deadlock
**Version Used**: f32205e548feffc6b763877866abbb47f1719a98

**Steps to Reproduce**:

1. On a background thread, call `Debug.Fail` (or `Debug.Assert(false)`)
2. On the UI thread, call `Debug.WriteLine`

**Expected Behavior**:

The use of `Debug.WriteLine` on the UI thread does not deadlock the integration test process.

**Actual Behavior**:

`Debug.WriteLine` blocks waiting for a lock, but the lock is held by the background thread for the duration of presenting the assertion window. The integration test process deadlocks.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #33599

### CODE REVIEW

File: src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs [hunk 1]
```diff
@@ -84,6 +84,8 @@ private void StartServiceCallback(object sender, EventArgs e)
         {
             if (_startMenuCmd.Enabled)
             {
+                IntegrationTestTraceListener.Install();
```
Comment (jaredpar) [inline src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs]: For unit tests we generally setup the trace listeners with app.config. We could just use a diff file for the integration tests.

https://github.com/dotnet/roslyn/blob/master/eng/config/test/Desktop/app.config#L9

Comment (sharwell) [reply src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs]: How would this get applied to devenv.exe?

Comment (jaredpar) [reply src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs]: Ah I see. Yeah that's not gonna work. I thought of this as inside the xunit process but it's in devenv.exe.

**Comment by jasonmalinowski** [comment]
@sharwell Thanks for doing this!

**Pull request merged**